ocs_ci/ocs/jenkins.py

```diff
@@ -0,0 +1,265 @@
+        timeout = timeout if timeout else 600
+        log.info(f"Waiting for the build to reach {status} state")
+        jenkins_builds = self.get_builds_obj()
+        for jenkins_build in jenkins_builds:
+            if jenkins_build.name not in self.build_completed:
+                try:
+                    wait_for_resource_state(
+                        resource=jenkins_build, state=status, timeout=timeout
+                    )
+                    self.build_completed.append(jenkins_build.name)
+                except ResourceWrongStatusException:
+                    error_msg = (
+                        f'{jenkins_build.name} did not reach to '
+                        f'{status} state after {timeout} sec\n'
+                    )
+                    log.error(error_msg)
+                    raise UnexpectedBehaviour(error_msg)
+
+    def get_jenkins_deploy_pods(self):
+        """
+        Get all jenkins deploy pods
+
+        Returns:
+            List: jenkins deploy pod objects list
+
+        """
+        return [
+            get_pod_obj(
+                pod
+            ) for pod in get_pod_name_by_pattern('deploy', namespace=constants.JENKINS_NAMESPACE)
+        ]
+
+    def get_builds_obj(self):
+        """
+        Get all jenkins builds
+
+        Returns:
+            List: jenkins deploy pod objects list
+
+        """
+        build_obj_list = []
+        build_list = self.get_build_name_by_pattern(
+            pattern=constants.JENKINS_BUILD, namespace=constants.JENKINS_NAMESPACE
+        )
+        for build_name in build_list:
+            ocp_obj = OCP(api_version='v1', kind='Build', namespace=constants.JENKINS_NAMESPACE)
+            ocp_dict = ocp_obj.get(resource_name=build_name)
+            build_obj_list.append(OCS(**ocp_dict))
+        return build_obj_list
+
+    def create_jenkins_pvc(self):
+        """
+        create jenkins pvc
+
+        """
+        pvc_obj = create_pvc(
+            pvc_name='dependencies', size='10Gi', sc_name='ocs-storagecluster-ceph-rbd',
+            namespace=constants.JENKINS_NAMESPACE
+        )
+        return pvc_obj
+
+    def create_app_jenkins(self):
+        """
+        create application jenkins
+
+        """
+        cmd = 'new-app --name=jenkins-ocs-rbd --template=jenkins-persistent-ocs'
+        self.ocp.exec_oc_cmd(command=cmd, out_yaml_format=False)
+
+    def start_build(self):
+        """
+        Start build on jenkins
+
+        """
+        log.info("Start build on jenkins")
+        cmd = f"start-build {constants.JENKINS_BUILD}"
+        self.ocp.exec_oc_cmd(command=cmd, out_yaml_format=False)
+
+    def setup_jenkins(self):
+        """
+        Setup jenkins
+
+        """
+        self.create_ocs_jenkins_template()
+        self.create_jenkins_pvc()
+        self.create_app_jenkins()
+        self.setup_jenkins_build_config()
+        self.wait_for_jenkins_deploy_status(status=constants.STATUS_COMPLETED)
+
+    def get_build_name_by_pattern(
```

Comment:
Similar generic function already exists in utils.py
def get_pod_name_by_pattern(
    pattern='client',
    namespace=None,
    filter=None,
):
Please check

Reply:
On "get_pod_name_by_pattern" function, the command "oc get pod" is static.
I need to use "oc get build" command.
![image](https://user-images.githubusercontent.com/61982127/85403774-9dfb3880-b566-11ea-8cfd-f26cad616ff9.png)